packages/database/src/core/AuthTokenProvider.ts

```diff
@@ -19,10 +19,27 @@ import { FirebaseApp } from '@firebase/app-types';
 import { FirebaseAuthTokenData } from '@firebase/app-types/private';
 import { log, warn } from './util/util';
 
+/**
+ * An interface for token fetchers.
+ */
+export interface AuthTokenProvider {
+  /**
+   * @param {boolean} forceRefresh
+   * @return {!Promise<FirebaseAuthTokenData>}
+   */
+  getToken(forceRefresh: boolean): Promise<FirebaseAuthTokenData>;
```

Comment:
The JS Doc comment adds little value. We can probably remove it, otherwise I'd suggest we document what `forceRefresh` is for.

Reply:
I'd prefer to remove it and add this documentation in a later PR.